Plugins/BuildNotifications.PluginInterfaces/Configuration/Options/Option.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using JetBrains.Annotations;
+
+namespace BuildNotifications.PluginInterfaces.Configuration.Options
+{
+    /// <summary>
+    /// Base class for an option.
+    /// </summary>
+    [PublicAPI]
+    public abstract class Option : IOption
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="nameTextId">Text id used for localizing the name of this option.</param>
+        /// <param name="descriptionTextId">Text id used for localizing the description of this option.</param>
+        protected Option(string nameTextId, string descriptionTextId)
+        {
+            NameTextId = nameTextId;
+            DescriptionTextId = descriptionTextId;
+        }
+
+        /// <summary>
+        /// Raises the <see cref="IsEnabledChanged" /> event.
+        /// </summary>
+        protected void RaiseIsEnabledChanged()
+        {
+            IsEnabledChanged?.Invoke(this, EventArgs.Empty);
+        }
+
+        /// <summary>
+        /// Raises the <see cref="IsLoadingChanged" /> event.
+        /// </summary>
+        protected void RaiseIsLoadingChanged()
+        {
+            IsLoadingChanged?.Invoke(this, EventArgs.Empty);
+        }
+
+        /// <summary>
+        /// Raises the <see cref="IsVisibleChanged" /> event.
+        /// </summary>
+        protected void RaiseIsVisibleChanged()
+        {
+            IsVisibleChanged?.Invoke(this, EventArgs.Empty);
+        }
+
+        /// <inheritdoc />
+        public bool IsLoading
+        {
+            get => _isLoading;
+            set
+            {
+                if (_isLoading == value)
+                    return;
+
+                _isLoading = value;
+                RaiseIsLoadingChanged();
+            }
+        }
+
+        /// <inheritdoc />
+        public string NameTextId { get; }
+
+        /// <inheritdoc />
+        public event EventHandler<EventArgs>? IsEnabledChanged;
+
+        /// <inheritdoc />
+        public event EventHandler<EventArgs>? IsVisibleChanged;
+
+        /// <inheritdoc />
+        public event EventHandler<EventArgs>? IsLoadingChanged;
```

Comment:
Imho this is much more clearer because I only have to subscribe to events I care about and don't have to make this switch all over the place when subscribing to a single PropertyChanged event.